src/Microsoft.ApiDesignGuidelines.Analyzers/Core/UseIntegralOrStringArgumentForIndexers.cs

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Analyzer.Utilities;
+
+namespace Microsoft.ApiDesignGuidelines.Analyzers
+{
+    /// <summary>
+    /// CA1043: Use Integral Or String Argument For Indexers
+    /// </summary>
+    public abstract class UseIntegralOrStringArgumentForIndexersAnalyzer : DiagnosticAnalyzer
```

Comment:
you can remove abstract and put [DiagnosticAnalyzer attribute here with both languages and then remove language specific one.